lib/middleware.js

```diff
@@ -597,7 +589,41 @@ module.exports = function middlewareConstructor() {
           return;
         }
 
-        req.session.publishUser = body;
+        req.user.publishId = body.id;
+        next();
+      });
+    },
+
+    /* Sets the project for the current request based on the project
+     * id provided as a parameter. The request's `user` property must
+     * be set by calling the `setUserIfTokenExists` middleware operation.
+     */
+    setProject: function(req, res, next) {
```

Comment:
This is poorly named, since it supports three cases: 1. No user, 2. user and no projectId,  3. user and projectId.

`maybeFetchProject` could work maybe?